@here/harp-mapview/lib/Tile.ts

```diff
@@ -607,11 +608,12 @@ export class Tile implements CachedResource {
         this.m_elevationRange.minElevation = elevationRange.minElevation;
         this.m_elevationRange.maxElevation = elevationRange.maxElevation;
         this.m_elevationRange.calculationStatus = elevationRange.calculationStatus;
+        this.elevateGeoBox();
 
         // Only elevate bounding box if tile has already been decoded and a maximum geometry height
         // is provided by the data source.
         if (this.m_maxGeometryHeight !== undefined) {
-            this.elevateBoundingBox();
+            this.updateBoundingBox();
```

Comment:
I know that `this.maxGeometryHeight` should be undefined in the case decoded tile provides fine tuned bounding box, but for a sake of simplicity it could be:
```
this.updateBoundingBox(this.m_decodedTile?.boundingBox);
```
Otherwise is look like we are loosing accurate bbox calculated in decoding phase.

Reply:
As you say, maxGeometryHeight and boundingBox are mutually exclusive, either you specify one or the other.
In decodedTile setter there's this code:
`        this.m_maxGeometryHeight = decodedTile.boundingBox
            ? undefined
            : decodedTile.maxGeometryHeight ?? 0;
`
Which ensures that if we have a bounding box, then the max geometry height is undefined. I've added an assertion inside this if to enforce that condition.

Reply:
Cool!